apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -0,0 +1,74 @@
+import { TRPCError } from "@trpc/server"
+import { z } from "zod"
+
+import { protectedProcedure, router } from "~/server/trpc"
+import { db } from "../database"
+
+export const resourceRouter = router({
+  getMetadataById: protectedProcedure
+    .input(
+      z.object({
+        resourceId: z.coerce.number().min(1),
+      }),
+    )
+    .query(async ({ input: { resourceId } }) => {
+      const resource = await db
+        .selectFrom("Resource")
+        .where("Resource.id", "=", resourceId)
+        .select(["permalink", "Resource.id", "Resource.title"])
+        .executeTakeFirst()
+
+      if (!resource) {
+        throw new TRPCError({ code: "NOT_FOUND" })
+      }
+
+      return resource
+    }),
+
+  getChildrenOf: protectedProcedure
+    .input(
+      z.object({
+        resourceId: z.union([z.coerce.number().min(1), z.null()]),
+      }),
+    )
+    .query(async ({ input: { resourceId } }) => {
+      let query = db
+        .selectFrom("Resource")
+        .select(["title", "permalink", "type", "id"])
+        .orderBy("type", "asc")
+        .orderBy("title", "asc")
+
+      if (resourceId === null) {
+        query = query.where("parentId", "is", null)
+      } else {
+        query = query.where("Resource.parentId", "=", resourceId)
+      }
+
+      return query.execute()
+    }),
+  move: protectedProcedure
+    .input(
+      z.object({
```

Comment:
should we move out validation schemas in a separate file for consistency?

Reply:
yes we should - lemme get on it

Reply:
done